packages/ai-chat-ui/src/browser/chat-input-widget.tsx

```diff
@@ -136,15 +136,41 @@ export class AIChatInputWidget extends ReactWidget {
         return this.navigationState.getNextPrompt();
     }
 
+    cycleMode(): void {
+        if (!this.receivingAgent || !this.receivingAgent.modes || this.receivingAgent.modes.length <= 1) {
+            return;
+        }
+        const currentIndex = this.receivingAgent.modes.findIndex(mode => mode.id === this.receivingAgent!.currentMode);
+        const nextIndex = currentIndex === -1 ? 1 : (currentIndex + 1) % this.receivingAgent.modes.length;
```

Comment:
if no index is currently found, should we not set 0?

Reply:
I set it to 1 so that it switches from the first item (index 0, selected by default) to the next item. Otherwise, the first `cycleMode` call wouldn't do anything, as it would cycle from -1 to 0 (first item).
This is just so that we can reuse the -1 that is returned from `findIndex`.